examples/go/with_proto/go/main_test.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	"os"
+	"testing"
+)
+
+func TestFlagMacros(t *testing.T) {
+	testCases := []struct {
+		envVar     string
+		validation func(string) bool
+		error      string
+	}{
+		{
+			envVar:     "FILE_NAME",
+			validation: func(expanded string) bool { return expanded == "main_test.go" },
```

Comment:
Because `$FileName$` requires the `main_test.go` file to be the active file for the test to pass, it can be confusing. I think we can avoid this by testing on something like $ProjectName$.

Reply:
Good point! Changed.